.devcontainer/devcontainer.json

```diff
@@ -25,10 +25,10 @@
 	],
 
 	// Use 'onCreateCommand' to run pre-build commands inside the codespace
-	"onCreateCommand": "${containerWorkspaceFolder}/.devcontainer/scripts/onCreateCommand.sh",
+	// "onCreateCommand": "${containerWorkspaceFolder}/.devcontainer/scripts/onCreateCommand.sh",
```

Comment:
These commands are important to the codespaces experience. They shouldn't be removed.

Reply:
They kick off a full release build of everything when you create a Codespace. It used all the CPU for quite a while even on the 16 core Codespace to build most of the subsets in the repo. Do we need to do that every time we start a Codespace or is there another aspect to it that I'm missing? I can uncomment these, but if we can avoid building the world on first run I'd like to.

Reply:
This is for pre-builts. That way when you create a new codespace, your repo is fully built and you can just start coding / running tests.
<img width="334" alt="image" src="https://user-images.githubusercontent.com/8291187/162035759-de287705-0fd5-4061-956d-7460d67d4dae.png">

See https://docs.github.com/en/codespaces/prebuilding-your-codespaces

Reply:
Interesting... those icons/UI elements didn't show up when I created my new Codespaces to test all this. Instead, it ran all these scripts when I created the container. Granted, I was creating a Codespace off my fork (my typical workflow) which may have complicated things. I'll uncomment these if it is supposed to be happening in an automated fashion _before_ the Codespace is spun up.

Reply:
I just created one, and it does appear to hang for a while after first create. Looking at `top -c`:

![image](https://user-images.githubusercontent.com/8291187/162038477-8d36c548-040e-4ae1-97a0-34a792e4249d.png)

I'm not sure what this is.

Reply:
I can't see the settings panel on the repo for this one, but do we have it all configured per this doc?
https://docs.github.com/en/codespaces/prebuilding-your-codespaces/configuring-prebuilds#configuring-a-prebuild

It's quite possible that because pre-builds are still in beta, that it hasn't fully rolled out, hence the inconsistency.

Reply:
>  Granted, I was creating a Codespace off my fork (my typical workflow) which may have complicated things

Yes - the prebuilds only work when creating the Codespace from the upstream dotnet/runtime repo.

> do we have it all configured per this doc?

I believe so. @agocke would know more for sure.

> It's quite possible that because pre-builds are still in beta,

They are. See https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-codespaces-prebuilds

>> Note: The ability to prebuild codespaces is currently in beta and subject to change.

Reply:
> Yes - the prebuilds only work when creating the Codespace from the upstream dotnet/runtime repo.

Is the expectation that we create a branch on dotnet/runtime or that we modify the remotes once the Codespace has been spun up? A little unfortunate either way, but the latter is a small enough hoop to jump through.

In either case, I put the scripts back since pre-builts are definitely worth having.

Reply:
@tanmayeekamath - thoughts on the expected workflow here for forks?

@josalem - I guess I always modify the remotes to add my fork as a remote and push there. One idea is using [Dotfiles](https://docs.github.com/en/enterprise-cloud@latest/codespaces/customizing-your-codespace/personalizing-codespaces-for-your-account#dotfiles) to do this automatically.

Reply:
The dotfiles repo is a good idea for doing that!